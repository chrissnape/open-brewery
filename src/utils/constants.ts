export const BREWERY = 'Brewery';
export const SEARCH_BREWERIES = 'Search Breweries';
