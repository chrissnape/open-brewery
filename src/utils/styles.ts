export const borderColour = '#ccc';
export const borderRadius = 10;
export const borderWidth = 1;
export const fontSize = 16;
export const padding = 10;
export const primaryColour = '#065535';
export const secondaryColour = '#e34f83';
