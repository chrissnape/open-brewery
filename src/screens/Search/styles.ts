import { StyleSheet } from 'react-native';
import { padding } from '../../utils/styles';

export default StyleSheet.create({
  cardWrapper: {
    marginBottom: padding,
  },
  container: {
    padding,
  },
});
