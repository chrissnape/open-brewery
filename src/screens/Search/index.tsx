import { StatusBar } from 'expo-status-bar';
import React, { FC, Fragment, useState, useEffect } from 'react';
import { View, ScrollView, Text, TouchableWithoutFeedback} from 'react-native';
import { BreweryCard, SearchBox } from '../../components';
import { BREWERY } from '../../utils/constants';
import { Brewery } from '../../utils/types';
import styles from './styles';

type Props = {
  breweries: Array<Brewery>;
  breweriesGetSuccess: boolean,
  breweriesGetFailure: boolean,
  favouriteBreweries: Array<number>;
  getBreweries: (city?: string) => void;
  navigation: any,
  queryCity: string | null,
  selectBrewery: (id: number) => void;
}

const SearchScreen: FC <Props> = ({ breweries, breweriesGetSuccess, breweriesGetFailure, favouriteBreweries, getBreweries, navigation, queryCity, selectBrewery }): JSX.Element => {
  const [queryString, setQueryString] = useState<string>('');
  const [isGetBreweries, setIsGetBreweries] = useState<boolean>(false);
  useEffect(() => {
    if (isGetBreweries === false) {
      getBreweries();
      setIsGetBreweries(true);
    }
  }, [isGetBreweries, getBreweries, setIsGetBreweries]);
  const renderContent = (): JSX.Element => {
    if (breweriesGetSuccess) {
      return (breweries.length > 0)
      ?  (
        <Fragment>
          {breweries.map((brewery: Brewery) => {
            const { id, city, name, state } = brewery;
            return (
              <View style={styles.cardWrapper} key={id}>
                <BreweryCard
                  city={city}
                  isFavourite={favouriteBreweries.includes(id)}
                  name={name}
                  onPress={() => {
                    selectBrewery(id);
                    navigation.navigate(BREWERY, { name });
                  }}
                  state={state}
                />
              </View>
            );
          })}
        </Fragment>
      )
      : (
        <Text>No breweries found</Text>
      );
    }
    if (breweriesGetFailure) {
      return (
        <Text>Failed to fetch breweries</Text>
      );
    }
    return (
      <Text>Fetching breweries</Text>
    );
  }
  return (
    <Fragment>
      <StatusBar style="light" />
      <View style={styles.searchBoxWrapper}>
        <SearchBox queryString={queryString} onChange={setQueryString} onSubmit={() => getBreweries(queryString)}/>
      </View>
      <View style={styles.queryWrapper}>
        <Text style={styles.queryString}>{queryCity || 'All'}</Text>
        {queryCity && (
          <TouchableWithoutFeedback onPress={() => {
            setQueryString('');
            getBreweries();
          }}>
            <Text style={styles.clearQueryString}>Clear</Text>
          </TouchableWithoutFeedback>
        )}
      </View>
      <ScrollView>
        <View style={styles.container}>
          {renderContent()}
        </View>
      </ScrollView>
    </Fragment>
  );
}

export default SearchScreen;
