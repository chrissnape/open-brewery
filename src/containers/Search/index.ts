import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import {
  getBreweries as getBreweriesAction,
  selectBrewery as selectBreweryAction,
} from '../../actions';
import { Search } from '../../screens';
import { AppState } from '../../store';

const mapStateToProps = (state: AppState) => {
  const { breweries } = state;
  return { ...breweries };
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  getBreweries: () => getBreweriesAction(dispatch),
  selectBrewery: (id: number) => selectBreweryAction(dispatch, id),
});

export default connect(mapStateToProps, mapDispatchToProps)(Search);
