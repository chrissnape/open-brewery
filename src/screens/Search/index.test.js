import React from 'react';
import renderer from 'react-test-renderer';
import Search from './index';
import { breweries } from '../../utils/mockObjects';

const props = {
  breweries: breweries,
  breweriesGetSuccess: true,
  breweriesGetFailure: false,
  favouriteBreweries: [],
  getBreweries: () => {},
  navigation: {},
  queryCity: null,
  selectBrewery: () => {},
}

describe('<Search />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Search { ...props } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});