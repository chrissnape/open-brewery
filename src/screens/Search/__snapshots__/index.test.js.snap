// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "padding": 10,
        "paddingBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        blurOnSubmit={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Search city..."
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#ccc",
            "borderWidth": 1,
            "flex": 1,
            "fontSize": 18,
            "marginRight": 10,
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text />
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#065535",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginRight": 10,
        }
      }
    >
      All
    </Text>
  </View>,
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 10,
                "width": "100%",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#065535",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                  }
                }
              >
                10 Barrel Brewing Co
              </Text>
              <Text>
                San Diego, California
              </Text>
            </View>
            <Text />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#ccc",
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 10,
                "width": "100%",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "#065535",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 5,
                  }
                }
              >
                2Kids Brewing Company
              </Text>
              <Text>
                San Diego, California
              </Text>
            </View>
            <Text />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
